addons/website_sale_coupon/tests/test_apply_pending_coupon.py

```diff
@@ -0,0 +1,91 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import TransactionCase, tagged
+from odoo.addons.website.tools import MockRequest
+from odoo.addons.website_sale_coupon.controllers.main import WebsiteSale
+
+import pprint
+
+pp = pprint.PrettyPrinter(indent=4)
+
+
+@tagged('-at_install', 'post_install')
+class TestSaleCouponApplyPending(TransactionCase):
+
+    def setUp(self):
+        super(TestSaleCouponApplyPending, self).setUp()
+
+        self.WebsiteSaleController = WebsiteSale()
+
+        self.website = self.env['website'].browse(1)
+        self.steve = self.env['res.partner'].create({
+            'name': 'Steve Bucknor',
+            'email': 'steve.bucknor@example.com',
+        })
+        self.empty_order = self.env['sale.order'].create({
+            'partner_id': self.steve.id
+        })
+        self.product = self.env['product.product'].create({
+            'name': 'Conference Chair',
+            'list_price': 100,
+            'taxes_id': False,
+        })
+
+        self.global_program = self.env['coupon.program'].create({
+            'name': '20% Discount',
+            'program_type': 'promotion_program',
+            'promo_code_usage': 'code_needed',
+            'promo_code': '20percent',
+            'reward_type': 'discount',
+            'discount_type': 'percentage',
+            'discount_percentage': 20,
+            'active': True,
+            'discount_apply_on': 'on_order',
+        })
+        self.coupon_program = self.env['coupon.program'].create({
+            'name': 'One Free Product',
+            'program_type': 'coupon_program',
+            'rule_min_quantity': 2.0,
+            'reward_type': 'product',
+            'reward_product_id': self.product.id,
+            'active': True,
+        })
+
+        self.coupon = self.env['coupon.coupon'].create({
+            'program_id': self.coupon_program.id,
+        })
+
+    def test_activate_coupon_with_existing_program(self):
+        order = self.empty_order
+
+        with MockRequest(self.env, website=self.website, sale_order_id=order.id) as request:
+            self.WebsiteSaleController.cart_update_json(self.product.id, set_qty=3)
+            self.assertEqual(order.amount_total, 300, "The order total should equal 300")
```

Comment:
```suggestion
            self.assertEqual(order.amount_total, 300, "The order total should equal 300: 3x100")
```
> The order total should equal 300

Yes this is obvious :) this is what the assert does, but why? Thats the point of the message